src/main/kotlin/ua/marchenko/artauction/common/annotation/profiling/beanPostProcessor/CustomProfilingAnnotationBeanPostProcessor.kt

```diff
@@ -0,0 +1,57 @@
+package ua.marchenko.artauction.common.annotation.profiling.beanPostProcessor
+
+import java.lang.reflect.Method
+import java.lang.reflect.Proxy
+import java.util.concurrent.TimeUnit
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.config.BeanPostProcessor
+import org.springframework.stereotype.Component
+import ua.marchenko.artauction.common.annotation.profiling.annotation.CustomProfiling
+import ua.marchenko.artauction.common.annotation.scheduled.beanPostProcessor.CustomScheduledAnnotationBeanPostProcessor
+
+@Component
+class CustomProfilingAnnotationBeanPostProcessor : BeanPostProcessor {
+
+    private val annotatedMethods: MutableMap<String, Map<Method, TimeUnit>> = mutableMapOf()
+
+    override fun postProcessBeforeInitialization(bean: Any, beanName: String): Any? {
+        val annotatedBeanClass = bean.javaClass
+        val annotatedBeanClassMethods: List<Method> =
+            annotatedBeanClass.methods.filter { it.isAnnotationPresent(CustomProfiling::class.java) }
+
+        if (annotatedBeanClassMethods.isNotEmpty()) {
+            val methodsWithTimeUnits = annotatedBeanClassMethods.associateWith { method ->
+                method.getAnnotation(CustomProfiling::class.java).timeUnit
+            }
+            annotatedMethods[beanName] = methodsWithTimeUnits
+        }
+        return bean
+    }
+
+    override fun postProcessAfterInitialization(bean: Any, beanName: String): Any? {
+        val methods = annotatedMethods[beanName] ?: return bean
+        val proxy =
+            Proxy.newProxyInstance(bean.javaClass.classLoader, bean.javaClass.interfaces) { _, method, args ->
+                val methodToProfile =
+                    methods.keys.find { it.name == method.name && it.parameters.contentEquals(method.parameters) }
+                if (methodToProfile != null) {
+                    val annotationTimeUnit = methods[methodToProfile] ?: TimeUnit.NANOSECONDS
+                    val beforeTime = System.nanoTime()
+
```

Comment:
This assign to proxy is huge. Consider refactoring this

Reply:
done